### ISSUE TITLE
Facility name does not receive focus on Facility locator detail page

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#25849]
## Issue Description
Issue identified during Screen reader usability session: Context lost when moving from search results to detail page because heading level did not receive focus

Also identified: Breadcrumb on this page does not follow the H1 rule
<details>
<summary> screenshot </summary>

![image](https://user-images.githubusercontent.com/55411834/121369953-8775fa80-c8f9-11eb-8f04-f2a464565581.png)
</details>

Name of VA facility is not in the tab workflow - focus moves from Home > Find Locations > Facility Details > Phone link
-  Per Angela and Josh, addressing this breadcrumb inconsistency should ensure facility name is presented to screen readers

**Adding this note from #20474:** 

> When I access a facilities details, the link to return to the search results is labeled "find location," implying that it returns me to the search screen without displaying search results.

---
## Tasks
- [x] Display the page title (h1) in the breadcrumb

## Acceptance Criteria
- [x] Veterans using a screen reader to visit a facility detail page hear the facility name in logical order
- [x] Issue to be validated by Josh and Angela for any additional fine tuning
- [x] Page H1 does *not* need to be focusable because the breadcrumb is satisfying that need
- [x] Breadcrumb gets focused in tab order and is announced to screenreaders

**Comment by gescobardev** [comment]
@joshkimux @AngelaFowler82 Would you be able to validate this ticket when you get a chance?
For reference: 
PR - https://github.com/department-of-veterans-affairs/vets-website/pull/17936
Review Instance - http://5d5777ccd352893003d643f80d27ee08.review.vetsgov-internal/find-locations/facility/vha_561GE

**Comment by AngelaFowler82** [comment]
I’m a little unclear as to what we’re validating here. When I click on a facility name from the search results, focus is placed at the beginning of the breadcrumb region.

**Comment by gescobardev** [comment]
Hi @AngelaFowler82, yes the focus is placed on the breadcrumbs.
Prior to the change I made, the breadcrumbs look like this:
<details> 
<summary>Screenshot (before)</summary>
<img width="617" alt="Screen Shot 2021-07-22 at 12 38 24 PM" src="https://user-images.githubusercontent.com/84030819/126675937-8b09c7b7-1812-406d-a5fe-3a8e5903b0e1.png">

</details>

With the change I made, it should now look like this: 
<details> 
<summary>Screenshot (after)</summary>
<img width="580" alt="Screen Shot 2021-07-22 at 12 39 56 PM" src="https://user-images.githubusercontent.com/84030819/126676159-7527aed7-fc40-481c-9076-b062ff618ff6.png">
</details>

The facility name should now be announced to the user, as well.

**Comment by AngelaFowler82** [comment]
My screen reader is unable to read screenshots, however I have no problem with the breadcrumb as it is structured. Having said that, the “find location” link still needs to be relabeled. The label implies that the user is being returned, not to the list of search results but to the search screen where they must enter their criteria. As to focus, it should be placed on the h1, as it is the beginning of the content the user is trying to access.

**Comment by gescobardev** [comment]
Apologies for the screenshots @AngelaFowler82. Here is the link for staging - https://staging.va.gov/find-locations/facility/vha_756QB
The task for this ticket was to change the breadcrumb from Facility Details to the actual name of the facility a user clicked on.

To address the re-labeling of "find locations" within the breadcrumb, we can look into creating a separate issue for this.
cc @mmiddaugh

**Comment by AngelaFowler82** [comment]
No worries. 

You will find the issue regarding the find locations link documented within the ticket. You will also find it documented here.

 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/20474

**Comment by gescobardev** [comment]
Oh perfect, thank you!
Am I okay to close this ticket now?

**Comment by gescobardev** [comment]
Oh perfect, thank you!
Am I okay to close this ticket now?

**Comment by AngelaFowler82** [comment]
As long as the focus management issue is addressed.

**Comment by joshkimux** [comment]
Gave this a quick look, it looks like the focus is not being sent to the `h1` which was the main issue within this ticket.

![image](https://user-images.githubusercontent.com/14154792/126705206-488a974f-b6c2-4e20-a398-04d2449e8382.png)

Should be a quick fix by adding a `tabindex="-1"` to the `h1` and sending focus to it should the user navigate to this page from the facility locator or any other page.

**Comment by joshkimux** [comment]
@gescobardev Let me know if you have any questions! More than happy to help over slack 😄

**Comment by gescobardev** [comment]
Hi @joshkimux, thank you for reviewing this 😁
I just followed the AC guidelines, part of it stated - " Page H1 does not need to be focusable because the breadcrumb is satisfying that need".

In addition to the breadcrumb change I made, I can also add `tabindex="-1"` to the h1.

**Comment by joshkimux** [comment]
@gescobardev @mmiddaugh @AngelaFowler82 Hey folks! Looks like there might have been a slight mixup in semantics when figuring out the a11y needs for this ticket.

@mmiddaugh , would it be possible for us to update the AC to:
<del>
- [ ] Page H1 does not need to be focusable because the breadcrumb is satisfying that need
- [ ] Breadcrumb gets focused in tab order and is announced to screenreaders
</del>

- [ ] Page H1 should receive focus using a `tabindex="-1"` in order to be announced to screen reader users when landing on the page

After confirming with Angela, the expected behavior from research is to have focus land on the `h1`-- apologies in advance for any confusion. We most likely got some words mixed up along the way 🙇

**Comment by mmiddaugh** [comment]
Thank you for the clarification @joshkimux 
@gescobardev will you please look at this again when you're feeling better?

**Comment by mpelzsherman** [comment]
Going forward, let's make a very clear distinction between _programmatic_ focus vs. _manual_ focus (tab order). Thanks.
